### PR TITLE
Update compatibility table and adding B2B for 2.4.1

### PR DESCRIPTION
**Pull request by dobooth**
## Purpose of this pull request

This pull request (PR) adds B2B to the Compatibility table.

## Affected DevDocs pages

- https://devdocs.magento.com/release/availability.html

### CODE REVIEW

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
We need to add new versions for all other modules. There's an internal wiki page that lists what those module versions are.

**Comment by dobooth** [comment]
Going to leave this as draft until just before release. Some extensions are marked as beta and I suspect will change at release.

**Comment by dshevtsov** [comment]
The following tests will be run for internal CI/CD testing, without an intention to merge.

**Comment by dshevtsov** [comment]
running tests

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
You're still missing Q4 entries (2.3.6/2.4.1) for Inventory and Product Recommendations.

**Review by hickskenh — APPROVED** [review]
only one change.  the rest looks to be formatted and outputs correctly.

**Comment by dobooth** [comment]
I hosed myself when I merged master instead of 2.4.1-develop. I forgot to add the work back in.

File: src/_data/mde.yml [hunk 1]
```diff
@@ -83,7 +109,21 @@ extensions:
           2.3.3: compatible
           2.3.4: compatible
           2.3.5-p1: compatible
-          2.4.0: supported
+          2.3.6: compatible
+          2.4.0: compatible
```
Comment (jeff-matthews) [inline src/_data/mde.yml:113]: Shouldn't 1.2.0 + 2.4.0 be "supported"?

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

File: src/_data/mde.yml (hunk 1 above)
Comment (dobooth) [reply src/_data/mde.yml:113]: Yes.

**Review by jeff-matthews — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**